dev/EnvironmentManager/ChangeTracker/EnvironmentVariableChangeTrackerHelper.h

```diff
@@ -0,0 +1,117 @@
+#if defined(__WINDOWSAPPRUNTIME_TEST_METADATA_H)
+    constexpr std::wstring_view frameworkPackageFamily{ L"Microsoft.WindowsAppRuntime.Framework-%i.%i_8wekyb3d8bbwe" };
+#else
+    constexpr std::wstring_view frameworkPackageFamily{ L"Microsoft.WindowsAppRuntime.%i.%i_8wekyb3d8bbwe" };
+#endif
+
+    std::unique_ptr<WCHAR> frameworkPackageFamilyName{ new WCHAR[PACKAGE_FAMILY_NAME_MAX_LENGTH + 1] };
+
+    // UINT16_MAX is big.  50 should be a reasonable MAJOR and MINOR.
+    constexpr uint16_t maxMajorToSearch{ 50 };
+    constexpr uint16_t maxMinorToSearch{ 50 };
+    bool foundAPackageInstalledForTheUser{};
+    for (uint16_t major = 1; major < maxMajorToSearch; major++)
+    {
+        for (uint16_t minor = 0; minor < maxMinorToSearch; minor++)
+        {
+            wsprintf(frameworkPackageFamilyName.get(), frameworkPackageFamily.data(), major, minor);
+
+            UINT32 packageCount = 0;
+            UINT32 packageNameLength = 0;
+            long returnCode{ FindPackagesByPackageFamily(frameworkPackageFamilyName.get(), PACKAGE_FILTER_DIRECT, &packageCount, nullptr, &packageNameLength, nullptr, nullptr)};
+
+            if (returnCode != ERROR_SUCCESS && returnCode != ERROR_INSUFFICIENT_BUFFER)
+            {
+                LOG_WIN32_MSG(returnCode, "Could error with trying to find the framework package.  Treating as 'Changes won't be tracked'");
+                return false;
+            }
+
+            if (packageNameLength > 0)
+            {
+                foundAPackageInstalledForTheUser = true;
+                break;
+            }
+        }
+
+        if (foundAPackageInstalledForTheUser)
+        {
+            break;
+        }
+    }
+
+    if (foundAPackageInstalledForTheUser)
+    {
+        return true;
+    }
+
+    // At this point no framework packages are installed for the current user.  However, a framework package might be installed
+    // machine wide.  Check that.  However, the process needs to be elevated to check.
+    if (!Security::IntegrityLevel::IsElevated())
+    {
+        return false;
+    }
+
+    winrt::Windows::Management::Deployment::PackageManager packageManager{};
+    auto currentArch{ AppModel::Identity::GetCurrentArchitecture() };
+    for (uint16_t major = 1; major < maxMajorToSearch; major++)
+    {
+        for (uint16_t minor = 0; minor < maxMinorToSearch; minor++)
+        {
+            wsprintf(frameworkPackageFamilyName.get(), frameworkPackageFamily.data(), major, minor);
+
+            for (auto package : packageManager.FindPackagesWithPackageTypes(frameworkPackageFamilyName.get(), winrt::Windows::Management::Deployment::PackageTypes::Framework))
+            {
+                if (package.Id().Architecture() == currentArch)
+                {
+                    return true;
+                }
+            }
+        }
+    }
+
+    return false;
+}
+
+inline bool ShouldChangesBeTracked(EnvironmentManager::Scope scope)
```

Comment:
More things to move to .cpp files

Reply:
I don't understand what you mean by this comment.  What should be moved to a cpp file?  ShouldChengesBeTracked?  Why?

Reply:
Feel free to unresolve if you believe items need to be moved to a .cpp file.